app/components/billing/payment.js

```diff
@@ -53,6 +58,38 @@ export default Component.extend({
     }
   }).drop(),
 
+  // amount_off and price are in cents
+  discountedPrice: computed('couponResult.{amount_off,percent_off}', 'selectedPlan.price', function () {
+    const price = Math.floor(this.selectedPlan.price / 100);
+    if (this.couponResult && this.couponResult.amount_off) {
+      const amountOff = this.couponResult.amount_off;
+      const discountedPrice = price - Math.floor(amountOff / 100);
+      return `$${discountedPrice}`;
+    } else if (this.couponResult && this.couponResult.percent_off) {
+      const percentageOff = this.couponResult.percent_off;
+      const discountedPrice = price - (price * percentageOff) / 100;
+      return `$${discountedPrice.toFixed(2)}`;
+    } {
+      return `$${price}`;
+    }
+  }),
+
+  validateCoupon: task(function* () {
+    try {
+      const result = yield this.api.get(`/coupons/${this.coupon}`);
+      this.set('couponResult', result);
+      this.set('isValidCoupon', result.valid);
```

Comment:
It seems a little strange to set `isValidCoupon` when it is already defined as `reads('couponResult.valid')`, but maybe there's a reason that I'm not thinking of?